src/components/course/data/utils.jsx

```diff
@@ -407,10 +412,329 @@ export function shortenString(str, maxLength, suffix, separator = ' ') {
+    DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED,
+    DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED,
+    DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED,
+    DISABLED_ENROLL_REASON_TYPES.COUPON_CODE_NOT_ASSIGNED,
+    DISABLED_ENROLL_REASON_TYPES.COUPON_CODES_EXPIRED,
+  ].includes(reasonType);
+
+  if (hasLimitsLearnMoreCTA) {
+    ensureConfig(['LEARNER_SUPPORT_SPEND_ENROLLMENT_LIMITS_URL']);
+    return (
+      <Button
+        as={Hyperlink}
+        className="text-center"
+        destination={getConfig().LEARNER_SUPPORT_SPEND_ENROLLMENT_LIMITS_URL}
+        target="_blank"
+        size="sm"
+        block
+      >
+        Learn about limits
+      </Button>
+    );
+  }
+
+  if (hasDeactivationLearnMoreCTA) {
+    ensureConfig(['LEARNER_SUPPORT_ABOUT_DEACTIVATION_URL']);
+    return (
+      <Button
+        as={Hyperlink}
+        className="text-center"
+        destination={getConfig().LEARNER_SUPPORT_ABOUT_DEACTIVATION_URL}
+        target="_blank"
+        size="sm"
+        block
+      >
+        Learn about deactivation
+      </Button>
+    );
+  }
+
+  if (hasContactAdministratorCTA) {
+    if (enterpriseAdminUsers?.length === 0) {
+      return null;
+    }
+    const adminEmails = enterpriseAdminUsers.map(({ email }) => email).join(',');
+    return (
+      <Button
+        as={MailtoLink}
+        className="text-center"
+        to={adminEmails}
+        target="_blank"
+        size="sm"
+        block
+      >
+        Contact administrator
+        <span className="sr-only">for help</span>
+      </Button>
+    );
+  }
+
+  return null;
+};
+
+export const getMissingApplicableSubsidyReason = ({
+  enterpriseAdminUsers,
+  catalogsWithCourse,
+  couponCodes,
+  couponsOverview,
+  customerAgreementConfig,
+  subscriptionLicense,
+  containsContentItems,
+  missingSubsidyAccessPolicyReason,
+}) => {
+  // Default disabled enrollment reason, assumes enterprise customer does not have any administrator users.
+  let reasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY_NO_ADMINS;
+  let userMessage;
+  const hasEnterpriseAdminUsers = enterpriseAdminUsers?.length > 0;
+
+  // If there are admin users, change `reasonType` to use the
+  // `DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY` message.
+  if (hasEnterpriseAdminUsers) {
+    reasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY;
+  }
+
+  const couponCodesDisabledEnrollmentReasonType = getCouponCodesDisabledEnrollmentReasonType({
+    catalogsWithCourse,
+    couponCodes,
+    couponsOverview,
+    hasEnterpriseAdminUsers,
+  });
+  const subscriptionsDisabledEnrollmentReasonType = getSubscriptionDisabledEnrollmentReasonType({
+    customerAgreementConfig,
+    catalogsWithCourse,
+    subscriptionLicense,
+    hasEnterpriseAdminUsers,
+  });
+  const enterpriseOffersDisabledEnrollmentReasonType = getEnterpriseOffersDisabledEnrollmentReasonType({
+    enterpriseOffers: customerAgreementConfig?.enterpriseOffers,
```

Comment:
D'oh, I missed this in yesterday's review. The `customerAgreementConfig` is related to subscriptions, not enterprise offers (i.e., `CustomerAgreement` == collection of `SubscriptionPlans` for a specific enterprise customer. I believe this will always return `undefined` as there is no `enterpriseOffers` property on the `customerAgreementConfig`.

It looks like `getMissingApplicableSubsidyReason` is not currently aware of the customer's `enterpriseOffers`; we'll want to pass `enterpriseOffers` as an argument to `getMissingApplicableSubsidyReason` so they can then be passed along to `getEnterpriseOffersDisabledEnrollmentReasonType` here.

Reply:
Good catch, I made that update!